packages/curve-ui-kit/src/features/connect-wallet/lib/hooks.ts

```diff
@@ -49,41 +50,46 @@ const useUseWagmi = (): boolean => {
 export const useWallet: UseConnectWallet = () => {
   const [{ wallet: onboardWallet, connecting: onboardConnecting }, onboardConnect, onboardDisconnect] =
     useOnboardWallet()
-  const [{ wallet: wagmiWallet, connecting: wagmiConnecting }, wagmiConnect, wagmiDisconnect] = useWagmiWallet()
+  const [{ wallet: wagmiWallet, connecting: wagmiConnecting }, wagmiConnect, wagmiDisconnect] = useWagmi()
   const [walletName, setWalletName] = useWalletName()
-  const useWagmi = useUseWagmi()
+  const shouldUseWagmi = useUseWagmi()
 
   const { wallet, provider } = useMemo(() => {
-    state.wallet = useWagmi ? wagmiWallet : onboardWallet && convertOnboardWallet(onboardWallet)
-    state.provider = state.wallet && new BrowserProvider(getRpcProvider(state.wallet))
+    state.wallet = shouldUseWagmi ? wagmiWallet : onboardWallet && convertOnboardWallet(onboardWallet)
+    state.provider = state.wallet && new BrowserProvider(getWalletProvider(state.wallet))
     return state
-  }, [onboardWallet, wagmiWallet, useWagmi])
+  }, [onboardWallet, wagmiWallet, shouldUseWagmi])
 
   const signerAddress = onboardWallet?.accounts[0]?.address
   const connect = useMemo(
     (): ((label?: string) => Promise<Wallet | null>) =>
-      useWagmi
-        ? wagmiConnect
+      shouldUseWagmi
+        ? (label?: string) => {
+            wagmiConnect(label)
+            return Promise.resolve(wagmiWallet)
+          }
         : (label?: string) =>
             onboardConnect({ ...(label && { autoSelect: { label, disableModals: true } }) }).then((wallets) =>
               convertOnboardWallet(wallets[0]),
             ),
-    [onboardConnect, useWagmi, wagmiConnect],
+    [onboardConnect, shouldUseWagmi, wagmiConnect, wagmiWallet],
   )
 
   const disconnect = useMemo(
     () =>
-      useWagmi
-        ? wagmiDisconnect
+      shouldUseWagmi
+        ? async () => {
+            wagmiDisconnect()
```

Comment:
@0xAlunara why no await here? :thinking: 
```suggestion
            await wagmiDisconnect()
```

Reply:
It wasn't async but I see you've already started using the async variant regardless